src/muya/lib/contentState/updateCtrl.js

```diff
@@ -1,12 +1,13 @@
 import { tokenizer } from '../parser/parse'
 import { conflict } from '../utils'
-import { CLASS_OR_ID, DEFAULT_TURNDOWN_CONFIG } from '../config'
+import { CLASS_OR_ID } from '../config'
 
 const INLINE_UPDATE_FRAGMENTS = [
   '^([*+-]\\s)', // Bullet list
   '^(\\[[x\\s]{1}\\]\\s)', // Task list
   '^(\\d+\\.\\s)', // Order list
-  '^\\s{0,3}(#{1,6})(?=\\s{1,}|$)', // ATX heading
+  '^\\s{0,3}(#{1,6})(?=\\s{1,}|$)', // ATX headings
+  '^\\s{0,3}(\\={3,}|\\-{3,})(?=\\s{1,}|$)', // Setext headings
```

Comment:
Three or more `-` characters are not recognized as thematic break any more. Type `foo`, then <kbd>Enter</kbd>, then `---`, <kbd>Enter</kbd> and `bar`. Result: `---` is not transformed in preview mode.

```markdown
foo

---

bar
```

Reply:
@fxha I didn't find that the `setext heading` syntax will affect `thematic break`, this issue fixed in the latest commit.